src/service.ts

```diff
@@ -387,6 +387,8 @@ export class Service {
       file.description = json.description;
       if (json.data) {
         file.setData(json.data);
+      } else if (json.data === null) {
+        file.setData("");
```

Comment:
It looks like you are trying to ensure that file data is either a valid string, or an array buffer, which is a great idea. To make sure this is always the case please add an assert in the `setData` function. This way all code can safely assume that `data` is not `null` or `undefined`.